Fix test result code instead of alias printed in test certificates detail screen (EXPOSUREAPP-7781)

jurajkusnier opened the pull request:
- Should work on WRU and other endpoints
- May not work on local cli

jurajkusnier committed:
update value sets init

BMItr committed:
Merge branch 'release/2.4.x' into feature/7781-update_value_sets

BMItter commented:
@jurajkusnier 
general question: why did you delete the valueSetRepo from the other viewmodel?
Wasn't it intended to be in both viewmodels?

jurajkusnier commented:
> @jurajkusnier
> general question: why did you delete the valueSetRepo from the other viewmodel?
> Wasn't it intended to be in both viewmodels?

I moved just the trigger for updating value sets, to be updated earlier (in the certificates screen). Once the values are updated they are stored in shared preferences so it's not necessary to call this trigger from the second screen.

harambasicluka committed:
Merge branch 'release/2.4.x' into feature/7781-update_value_sets

BMItter commented:
> > @jurajkusnier
> > general question: why did you delete the valueSetRepo from the other viewmodel?
> > Wasn't it intended to be in both viewmodels?
> 
> I moved just the trigger for updating value sets, to be updated earlier (in the certificates screen). Once the values are updated they are stored in shared preferences so it's not necessary to call this trigger from the second screen.

Yes I got it.
Could you please check whether the Viewmodel is called BEFORE the testCertificat && BEFORE the Vaccination Part?
If its not the case we will move the bug around.

jurajkusnier commented:
> > > @jurajkusnier
> > > general question: why did you delete the valueSetRepo from the other viewmodel?
> > > Wasn't it intended to be in both viewmodels?
> > 
> > 
> > I moved just the trigger for updating value sets, to be updated earlier (in the certificates screen). Once the values are updated they are stored in shared preferences so it's not necessary to call this trigger from the second screen.
> 
> Yes I got it.
> Could you please check whether the Viewmodel is called BEFORE the testCertificat && BEFORE the Vaccination Part?
> If its not the case we will move the bug around.

Yes, as we moved Vaccinations to the Certificates tab, I don't see any way how to access Vaccinations without opening the Certificates tab at least once.

jurajkusnier committed:
Merge branch 'release/2.4.x' into feature/7781-update_value_sets

BMItr committed:
Merge branch 'release/2.4.x' into feature/7781-update_value_sets

jurajkusnier committed:
revert changes in VaccinationListViewModel

jurajkusnier committed:
Merge remote-tracking branch 'origin/feature/7781-update_value_sets' into feature/7781-update_value_sets

jurajkusnier commented:
> > > > @jurajkusnier
> > > > general question: why did you delete the valueSetRepo from the other viewmodel?
> > > > Wasn't it intended to be in both viewmodels?
> > > 
> > > 
> > > I moved just the trigger for updating value sets, to be updated earlier (in the certificates screen). Once the values are updated they are stored in shared preferences so it's not necessary to call this trigger from the second screen.
> > 
> > 
> > Yes I got it.
> > Could you please check whether the Viewmodel is called BEFORE the testCertificat && BEFORE the Vaccination Part?
> > If its not the case we will move the bug around.
> 
> Yes, as we moved Vaccinations to the Certificates tab, I don't see any way how to access Vaccinations without opening the Certificates tab at least once.

I reverted changes in `VaccinationListViewModel` after discussion regarding concerns for potential future changes in screen flow

jurajkusnier committed:
Merge branch 'release/2.4.x' into feature/7781-update_value_sets

# Conflicts:
#	Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/test/ui/CertificatesViewModel.kt

mtwalli commented on Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/test/ui/CertificatesViewModel.kt:
```diff
@@ -19,17 +21,25 @@ import de.rki.coronawarnapp.covidcertificate.vaccination.ui.cards.ImmuneVaccinat
 ) : CWAViewModel() {
 
+    init {
+        valueSetsRepository.triggerUpdateValueSet(languageCode = context.getLocale())
```
does it have to be `context.getLocale()` , can we use `Locale.getDefault()` as everywhere else in the App?

jurajkusnier replied:
I guess we can use both variants. I used the same as we are using in the VaccinationListViewModel in the same method call.

harambasicluka committed:
Merge branch 'release/2.4.x' into feature/7781-update_value_sets

jurajkusnier committed:
Merge branch 'release/2.4.x' into feature/7781-update_value_sets
